From b2a86a11dae4ed77e4650fc5b00ceea1bfdb5bd0 Mon Sep 17 00:00:00 2001
From: Tobias Jungel <tobias.jungel@gmail.com>
Date: Wed, 4 Oct 2017 14:19:33 +0200
Subject: [PATCH 4/5] check for protobuf 3.4.0
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

in case protobuf is installed on the system and pkg-config can be used
the version check has to be against 3.4.0. Otherwise you run into:

./src/compiler/php_generator_helpers.h: In function ‘grpc::string grpc_php_generator::GetPHPServiceFilename(const FileDescriptor*, const ServiceDescriptor*, const string&)’:
./src/compiler/php_generator_helpers.h:51:23: error: ‘const class google::protobuf::FileOptions’ has no member named ‘has_php_namespace’; did you mean ‘has_csharp_namespace’?
   if (file->options().has_php_namespace()) {
                       ^~~~~~~~~~~~~~~~~
                       has_csharp_namespace
./src/compiler/php_generator_helpers.h:52:39: error: ‘const class google::protobuf::FileOptions’ has no member named ‘php_namespace’; did you mean ‘csharp_namespace’?
     oss << ReplaceAll(file->options().php_namespace(), "\\", "/");
                                       ^~~~~~~~~~~~~
                                       csharp_namespace
src/compiler/php_generator.cc: In function ‘grpc::string grpc_php_generator::{anonymous}::PackageName(const FileDescriptor*)’:
src/compiler/php_generator.cc:47:23: error: ‘const class google::protobuf::FileOptions’ has no member named ‘has_php_namespace’; did you mean ‘has_csharp_namespace’?
   if (file->options().has_php_namespace()) {
                       ^~~~~~~~~~~~~~~~~
                       has_csharp_namespace
src/compiler/php_generator.cc:48:28: error: ‘const class google::protobuf::FileOptions’ has no member named ‘php_namespace’; did you mean ‘csharp_namespace’?
     return file->options().php_namespace();
                            ^~~~~~~~~~~~~
                            csharp_namespace
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 6582b3bdd8..ba34c4bcc1 100644
--- a/Makefile
+++ b/Makefile
@@ -493,7 +493,7 @@ ifeq ($(HAS_PKG_CONFIG),true)
 OPENSSL_ALPN_CHECK_CMD = $(PKG_CONFIG) --atleast-version=1.0.2 openssl
 OPENSSL_NPN_CHECK_CMD = $(PKG_CONFIG) --atleast-version=1.0.1 openssl
 ZLIB_CHECK_CMD = $(PKG_CONFIG) --exists zlib
-PROTOBUF_CHECK_CMD = $(PKG_CONFIG) --atleast-version=3.0.0 protobuf
+PROTOBUF_CHECK_CMD = $(PKG_CONFIG) --atleast-version=3.4.0 protobuf
 CARES_CHECK_CMD = $(PKG_CONFIG) --atleast-version=1.11.0 libcares
 else # HAS_PKG_CONFIG
 
-- 
2.13.6

